shoes-core/lib/shoes/common/image_handling.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+class Shoes
+  module Common
+    module ImageHandling
+      def absolute_file_path(path)
+        if Pathname(path).absolute?
+          search_for(File.basename(path), File.dirname(path))
+        else
+          search_for(path, *default_search_paths)
+        end
+      end
+
+      def search_for(path, *locations)
+        found = locations.map { |dir| File.join(dir, path) }
+                         .find { |candidate| File.exist?(candidate) }
+
+        unless found
+          raise FileNotFoundError, "#{path} not found. Searched #{locations.join(',')}"
+        end
+
+        found
+      end
+
+      def default_search_paths
+        [Dir.pwd, Shoes.configuration.app_dir, File.join(Shoes::DIR, "static")]
```

Comment:
At first blush I thought memoizing might be nice--just to express that this doesn't change--but then I considered that `Dir.pwd` _could_ change. Imagine building an image viewing app that works on directories... you could just `Dir.chdir` into those and then use the unexpanded filenames, which seems kinda nice.